nikola/filters.py

```diff
@@ -397,3 +397,39 @@ def _normalize_html(data):
 
 
 normalize_html = apply_to_text_file(_normalize_html)
+
+
+@_ConfigurableFilter(xpath_list='HEADER_PERMALINKS_XPATH_LIST')
+@apply_to_text_file
+def add_header_permalinks(data, xpath_list=None):
+    """Post-process HTML via lxml to add header permalinks Sphinx-style."""
+    doc = lxml.html.document_fromstring(data)
+    # Get language for slugify
+    try:
+        lang = doc.attrib['lang']  # <html lang="…">
+    except KeyError:
+        # Circular import workaround (utils imports filters)
+        from nikola.utils import LocaleBorg
+        lang = LocaleBorg().current_lang
+
+    if not xpath_list:
+        xpath_list = ['*//div[@class="e-content entry-content"]//{hx}']
+    for xpath_expr in xpath_list:
+        for hx in ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']:
+            nodes = doc.findall(xpath_expr.format(hx=hx))
```

Comment:
In case `xpath_expr` does *not* contain `hx`, wouldn't this add a permalink six times for every matching node?